test/Toggle/Segment/SegmentPriorityArrangerTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Clearbooks\Labs\Client\Toggle\Segment;
+
+use Clearbooks\Labs\Client\Toggle\Entity\SegmentStub;
+
+class SegmentPriorityArrangerTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var SegmentPriorityArranger
+     */
+    private $segmentPriorityArranger;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->segmentPriorityArranger = new SegmentPriorityArranger();
+    }
+
+    /**
+     * @test
+     */
+    public function GivenEmptyArray_ExpectEmptyArray()
+    {
+        $segments = [ ];
+        $orderedSegments = $this->segmentPriorityArranger->orderSegmentsByPriority( $segments );
+        $this->assertEquals( $segments, $orderedSegments );
+    }
+
+    /**
+     * @test
+     */
+    public function GivenUnorderedSegments_ExpectOrderedListByPriority()
+    {
+        $segments = [
+            new SegmentStub( 5, 5 ),
+            new SegmentStub( 4, 4 ),
+            new SegmentStub( 3, 3 ),
+            new SegmentStub( 2, 2 ),
+            new SegmentStub( 1, 1 )
+        ];
+
+        $shuffledSegments = $segments;
```

Comment:
No, the next line does :D This just copies the array. Assigning variables by reference has to be explicit in PHP, it copies by default.